### PR TITLE
Add nginx.enabled flag

### PR DESCRIPTION
**Pull request by Smirl**
# Summary

Relates to #32 

- Add `.Values.nginx.enabled` which defaults to true
- Can be disabled if using your own ingress solution
  - This could be a set of Ingress objects as per the issue
  - Could be a service mesh object such as an istio `VirtualService`
- Also removes a workaround port in the custom-actions app as the comment suggests it is only for nginx healthchecks

### CODE REVIEW

**Comment by sara-tagger** [comment]
Thanks for submitting a pull request 🚀 @amn41 will take a look at it as soon as possible ✨

**Comment by tczekajlo** [comment]
@Smirl Thanks a lot for your PR, looks great. There is only one thing that caught my eye, in the `templates/ingress.yaml` we use nginx service and as a backend. The ingress should take into account that Nginx is disabled.

**Comment by Smirl** [comment]
Thanks for taking a look @tczekajlo 

Do you think that I should just disable the ingress if nginx is disabled for now? The other option is to implement the 4 `Ingress` objects that @rafi suggested in #32. I don't mind copy pasting those (with some template modifications) however I don't have an ingress controller set up to test (we use istio). I could probably get nginx ingress up and running quickly.

**Comment by tczekajlo** [comment]
@Smirl it would be great to implement suggestions from #32, but it has to be tested :D If you feel that is something that you can do then go for it otherwise we can go for option one and disable the ingress if Nginx is disabled too.

Thanks in advance! 🙇

**Comment by Smirl** [comment]
Hey, I have pushed a commit to disable the ingress when nginx is disabled for now.

I did deploy the ingress objects above and they worked well. However, a few points:

- the `rasa-x-chat` Ingress which rewrites to different environments isn't really needed as `/api/chat` in rasa-x will route you to the correct environment. I can only think this was done as an optimisation however most channels will go to rasa directly (i.e. `/core/webhooks/slack/webhook`)
- these objects have nginx-ingress specific annotations so when I was templating them it didn't feel right
  - perhaps a more generic values.yaml to specify multiple ingresses, rather than the single object.
  - similar to the current `.Values.ingress.hosts[]`
  - perhaps `.Values.ingress.ingresses[]`?
- I didn't test the `/socket.io` however could it be removed and URLs pointed to `/core/socket.io`?

In general, the `/app/` and `/core/` don't seem to be referred to in the docs much.
- app is just for custom actions which is reached from within the cluster by the rasa deployments
- core is how you would expose slack, etc. to the world, so might just need to docs to be updated.

File: charts/rasa-x/templates/ingress.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{{- if .Values.ingress.enabled -}}
+{{- if and .Values.ingress.enabled .Values.nginx.enabled -}}
```
Comment (tczekajlo) [inline charts/rasa-x/templates/ingress.yaml:1]: Cool, only one small thing. Can you add a comment in the value file that `ingress.enabled` requires `nginx.enabled=true` in order to work?

https://github.com/RasaHQ/rasa-x-helm/blob/967d470db7b530151e29dda71dbbe0c0d092f15e/charts/rasa-x/values.yaml#L401

Comment (Smirl) [reply charts/rasa-x/templates/ingress.yaml:1]: Thanks. I added a small comment :)

**Comment by tczekajlo** [comment]
@Smirl Thanks a lot for the contribution 🎉

**Pull request merged**